app/code/Magento/ConfigurableProduct/Model/ResourceModel/Product/LinkedProductSelectBuilder.php

```diff
@@ -3,9 +3,10 @@
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
+declare(strict_types=1);
+
 namespace Magento\ConfigurableProduct\Model\ResourceModel\Product;
 
-use Magento\Catalog\Model\ResourceModel\Product\BaseSelectProcessorInterface;
```

Comment:
this changes the constructor argument type which would break installations where an instance of the original interface is injected into this class

Reply:
fixed